compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ServiceDesugar.java

```diff
@@ -201,7 +201,8 @@ void engageCustomServiceDesugar(BLangService service, SymbolEnv env) {
     }
 
     private void engageCustomResourceDesugar(BLangFunction functionNode, SymbolEnv env) {
-        httpFiltersDesugar.addHttpFilterStatementsToResource(functionNode, env);
-        httpFiltersDesugar.addCustomAnnotationToResource(functionNode, env);
+        if (HttpFiltersDesugar.isHttpResource(functionNode)) {
```

Comment:
Do we really need this check? This method is called for resource functions only.

Reply:
Yes, but this is called for both HTTP and WebSocket resource functions. Since the service typing support is not available yet, this method is used to check the first function parameter to distinguish it.